src/antidote_dc_manager.erl

```diff
@@ -45,41 +56,23 @@
 -include("inter_dc_repl.hrl").
 -include_lib("kernel/include/logger.hrl").
 
--export([create_dc/1,
-         get_connection_descriptor/0,
-         subscribe_updates_from/1]
-       ).
+-export([
+    create_dc/1,
+    add_nodes_to_dc/1,
+    get_connection_descriptor/0,
+    subscribe_updates_from/1
+]).
+
 
 %% Build a ring of Nodes forming a data center
--spec create_dc([node()]) -> ok.
-create_dc(Nodes) ->
-    ?LOG_INFO("Creating DC ring with nodes ~p", [Nodes]),
-    %% Ensure each node owns 100% of it's own ring
-    _ = [[Node] = owners_according_to(Node) || Node <- Nodes],
-    %% Join nodes
-    [Node1 | OtherNodes] = Nodes,
-    case OtherNodes of
-        [] ->
-            %% no other nodes, nothing to join/plan/commit
-            ok;
-        _ ->
-            %% ok do a staged join and then commit it, this eliminates the
-            %% large amount of redundant handoff done in a sequential join
-            [staged_join(Node, Node1) || Node <- OtherNodes],
-            plan_and_commit(Node1),
-            try_nodes_ready(Nodes, 3, 500)
-    end,
-
-    ok = wait_until_nodes_ready(Nodes),
-
-    %% Ensure each node owns a portion of the ring
-    wait_until_nodes_agree_about_ownership(Nodes),
-    ok = wait_until_no_pending_changes(Nodes),
-    wait_until_ring_converged(Nodes),
-    ok = wait_until(hd(Nodes), fun wait_init:check_ready/1),
-    %% starts metadata services needed for intra-dc communication
-    ok = inter_dc_manager:start_bg_processes(stable_time_functions),
-    ok.
+-spec add_nodes_to_dc([node()]) -> ok | {error, ring_not_ready}.
+add_nodes_to_dc(Nodes) ->
+    %% check if ring is ready first
+    case riak_core_ring:ring_ready() of
+        true -> join_new_nodes(Nodes);
+        _ -> {error, ring_not_ready}
```

Comment:
Why not wait until the ring is ready? 
Or are there cases where it will never be ready after some waiting?

Reply:
Ring modifications are rare and should be planned with care. I'd rather throw an error to the client indicating a ring modification of some other user at the same time rather than waiting for the current ring to be ready and getting maybe unexpected ring changes for both users.

It would be no problem to add a sleep and retry the function call, though.